controllers/system/system_controller.go

```diff
@@ -1102,6 +1102,19 @@ func (r *SystemReconciler) ReconcileRequired(instance *starlingxv1.System, spec
 		return err, false
 	}
 
+	var res starlingxv1.CertificateList
+	certificateMap := make(map[string]bool)
+
+	for _, eachCertificate := range *spec.Certificates {
+		certificateMap[eachCertificate.Secret] = true
```

Comment:
Any reason why comparing based on the secret rather than the signature?

Reply:
Because by that point spec doesn't get the filled with the corresponding signatures
Signatures will be empty values in the spec

Reply:
Suggest to add a comment in the new method to explain that.